python/pyarrow/_flight.pyx

```diff
@@ -704,7 +705,7 @@ cdef class FlightEndpoint(_Weakrefable):
     cdef:
         CFlightEndpoint endpoint
 
-    def __init__(self, ticket, locations):
+    def __init__(self, ticket, locations, expiration_time, app_metadata):
```

Comment:
Same comment as the previous PR about making these optional for backwards compatibility assuming we can figure out the cython issue.

Reply:
done, thanks!